core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -324,14 +337,62 @@ public JDBCUserStoreManager(RealmConfiguration realmConfig, Map<String, Object>
         }
 
         initUserRolesCache();
-
+        if (hashProvider == null) {
+            hashProvider = initializeHashProvider(realmConfig.getUserStoreProperties());
+        }
         if (log.isDebugEnabled()) {
             log.debug("Ended " + System.currentTimeMillis());
         }
         /* Initialize user roles cache as implemented in AbstractUserStoreManager */
 
     }
 
+    /**
+     * Initializes the HashProvider.
```

Comment:
`Initialize the HashProvider according to the given user store properties.`

Reply:
resolved